cmd/utils.go

```diff
@@ -30,6 +33,39 @@ func init() {
 	flag_.CommandLine.Usage = PrintDefaults
 }
 
+type SignalCallback func()
+type SignalHandler struct {
+	ch        chan os.Signal
+	listeners []net.Listener
+	callbacks []SignalCallback
+	osSignal  os.Signal
+}
+
+func NewSignalHandler(handledSignal os.Signal) (*SignalHandler, error) {
+	return &SignalHandler{ch: make(chan os.Signal), osSignal: handledSignal}, nil
+}
+
+func (handler *SignalHandler) AddListener(listener net.Listener) {
+	handler.listeners = append(handler.listeners, listener)
+}
+
+func (handler *SignalHandler) AddCallback(callback SignalCallback) {
+	handler.callbacks = append(handler.callbacks, callback)
+}
+
+// Register should be called as goroutine
+func (handler *SignalHandler) Register() {
+	signal.Notify(handler.ch, handler.osSignal)
+	<-handler.ch
```

Comment:
is it channel? https://tour.golang.org/concurrency/2

where it sends value from `handler.ch`?

Reply:
https://golang.org/pkg/os/signal/#Notify 
here we pass channel and block by reading until app receive signal